src/components/CardCategory/useCardCategory.ts

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { useSelector, useDispatch } from "react-redux";
+import {useEffect} from 'react'
+import {actions,selectorsCategory} from '../../store/sliceCategory'
+import { AppDispatch } from 'store/store';
+
+
+export const useCardCategory = (id:string) => {
+  const dispatch=useDispatch<AppDispatch>();
+  const category=useSelector(selectorsCategory.getCategory)
```

Comment:
```suggestion
  const { items } = useSelector(selectorsCategory.getCategory)
```
можно сразу Items доставать, а вообще это селектор должен сразу отдать Items готовый, чтобы компонент не знал о структуре